### PR TITLE
Correct typos and grammar issues in Javadoc

### PR DESCRIPTION
**Pull request by richdouglasevans**
Some minor typos and suchlike that I spotted while reading through the codebase.

### CODE REVIEW

File: src/main/java/com/greghaskins/spectrum/AbstractSupplyingHook.java [hunk 1]
```diff
@@ -6,8 +6,9 @@
 import org.junit.runner.notification.RunNotifier;
 
 /**
- * A base class for supplying hooks to use. Override before or after. Return the singleton
- * value from the before method.
+ * A base class for {@link SupplyingHook hooks that supply a value}.
+ *
+ * <p>Override {@link #before} or {@link #after}. Return the singleton value from the before method.
```
Comment (richdouglasevans) [inline src/main/java/com/greghaskins/spectrum/AbstractSupplyingHook.java]: The existing opening sentence read ambiguously: it's not for supplying actual hooks, it's for hooks that supply a value.

While I was there I hyperlinked to the referenced methods to help them "pop" and to call out that they're not just words but method names.

File: src/main/java/com/greghaskins/spectrum/Spectrum.java [hunk 2]
```diff
@@ -250,22 +250,24 @@ public static void beforeEach(final com.greghaskins.spectrum.Block block) {
   }
 
   /**
-   * Declare a {@link Block} to be run after each spec in the current suite.
+   * Declare a {@link com.greghaskins.spectrum.Block Block} to be run
+   * after each spec in the current suite.
```
Comment (richdouglasevans) [inline src/main/java/com/greghaskins/spectrum/Spectrum.java]: The existing hyperlink referenced a ~deprecated~ class: I merely updated the reference to the class that is not deprecated.

File: src/main/java/com/greghaskins/spectrum/Spectrum.java [hunk 3]
```diff
@@ -331,7 +333,7 @@ public static void aroundAll(ThrowingConsumer<com.greghaskins.spectrum.Block> co
    * @param <T> The type of value
    *
    * @param supplier {@link com.greghaskins.spectrum.ThrowingSupplier} function that either
-   *        generates the value, or throws a `Throwable`
+   *        generates the value, or throws a {@link Throwable}
```
Comment (richdouglasevans) [inline src/main/java/com/greghaskins/spectrum/Spectrum.java]: Backticks are code markers in Markdown only.

File: src/main/java/com/greghaskins/spectrum/SupplyingHook.java [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import java.util.function.Supplier;
 
 /**
- * For when adding a hook that supplies a value within running itself. The hook will consume
+ * A hook that supplies a value within a running test. The hook will consume
```
Comment (richdouglasevans) [inline src/main/java/com/greghaskins/spectrum/SupplyingHook.java]: The existing opening sentence read a bit _Engrish_ (sic).

**Comment by ashleyfrieze** [comment]
:+1:

**Comment by richdouglasevans** [comment]
Cheers for the :+1: Ash! I'm hoping to make pull requests with more meat in the near future :smile:

**Comment by greghaskins** [comment]
Looks great, thanks @richdouglasevans !

**Pull request merged**